src/main/java/de/tum/in/www1/artemis/service/exam/ExamService.java

```diff
@@ -815,7 +815,7 @@ private boolean hasNonEmptySubmission(Set<Submission> submissions, Exercise exer
         }
         else if (exercise instanceof FileUploadExercise) {
             FileUploadSubmission textSubmission = (FileUploadSubmission) submissions.iterator().next();
-            return textSubmission.getFilePath() != null && !textSubmission.getFilePath().isEmpty();
+            return !textSubmission.getFilePaths().isEmpty() && textSubmission.getFilePaths().stream().noneMatch(String::isEmpty);
```

Comment:
I've checked this out and the fileUploadSubmission.getFilePaths() URI's come from FilePathService::publicPathForActualPath. I don't think this method has a path that returns an empty string but if the user client submits a file that starts with an unknown directory it will return null. Still I think it's a good idea to apply your suggestion just in case there might be empty strings for some reason. I would also filter out all null entries right where publicPathForActualPath is called in FileUploadSubmissionService::storeFiles since (especially) stream operations like map on null entries might cause an exception unexpectedly somewhere downstream when called on nulls.